File: 2021/11/27/average_salary.js
/**
 * https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/
 * @author Andrew Jarombek
 * @since 11/27/2021
 */

const average = function(salary) {
    let minimum = 1_000_001;
    let maximum = 99;
    let sum = 0;

    for (let num of salary) {
        sum += num;

        if (num > maximum) {
            maximum = num;
        }

        if (num < minimum) {
            minimum = num;
        }
    }

    return (sum - minimum - maximum) / (salary.length - 2)
};

File: 2022/08/14/average_salary.py
"""
https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/
Author: Andrew Jarombek
Date: 8/14/2022
"""

from typing import List


def average(salary: List[int]) -> float:
    total = salary[0]
    minimum = total
    maximum = total

    for value in salary[1:]:
        total += value
        minimum = min(minimum, value)
        maximum = max(maximum, value)

    return (total - minimum - maximum) / (len(salary) - 2)


if __name__ == '__main__':
    assert average([4000, 3000, 1000, 2000]) == 2500
